src/libstore/globals.hh

```diff
@@ -524,6 +524,25 @@ public:
     Setting<bool> sandboxFallback{this, true, "sandbox-fallback",
         "Whether to disable sandboxing when the kernel doesn't allow it."};
 
+    Setting<bool> dropSupplementaryGroups{this, getuid() == 0, "drop-supplementary-groups",
+        R"(
+          Whether to drop supplementary groups when building with sandboxing.
+          This is normally a good idea if we are root and have the capability to
+          do so.
```

Comment:
When is it not? I think we should warn about negative groups (if that's the right name for it); groups that when you're a member, you have *fewer* capabilities rather than more.

It would guess for it to be rare that the daemon would start as root but have a negative group, but I don't think we are free to assume just assume that and be quiet about it here.

Reply:
We could document our opinion, "open letter" style.

> contravariant

That corresponds to my "negative". Not sure which is better. Something "anti-..." could fit and be more explicit. Needs more thought or more reading, or anything will do.

Reply:
Yeah I think postive/negative is better than co-/contrainvariant :), but then there is "invariant" too, and what does that map to? "Neutral" sounds more innocuous, like "bivariant" to me.

Reply:
Going back to the original question, I am actually fine with Nix assuming all groups are "positive", and always trying to drop groups. Non-positive ones are gross to me and I have little interest in trying to support them.

The real question question is if that operation *fails*, is it a fatal error or not? That is what feels to me like like a question that ultimately needs to be in the user's not Nix's hand.

(To simply unblock myself writing tests, and also for backwards compat since Nix didn't try to do the operation as non-root before, I made the option on whether to attempt the option. I have issue to instead make the setting about what to do if the operation fails with EPERM or whatever. Non-root could attempt the operation anyways, after all, it's possible to have the `CAP_SETGID` without being root so it might in fact succeed.)